apps/web/app/my-reviews/page.tsx

```diff
@@ -0,0 +1,141 @@
+"use client";
+
+import { MyReviewCard } from "@components/ui/cards/my-review-card";
+import SessionModal from "@components/ui/session-modal";
+import { Accordion, Badge, Center, Grid, Loader, Paper, Stack, Text } from "@mantine/core";
+import { useDisclosure } from "@mantine/hooks";
+import { APPROVED, BASE_API_ENDPOINT, ERROR, INFO, PENDING, REJECTED, SUCCESS } from "@utils/constants";
+import fetcher from "@utils/fetcher";
+import { useAuth } from "hooks/use-auth";
+import { useEffect, useState } from "react";
+import type { AccordionItems, MyReviewsData } from "types";
+import type { Session } from "@supabase/supabase-js";
+
+const initialEmptyPosts: MyReviewsData[] = [];
+
+export default function MyReviews(): JSX.Element {
+	const [sessionData, setSessionData] = useState<Session | null>();
+	const [isLoading, setIsLoading] = useState<boolean>(true);
+	const [myReviewsData, setMyReviewsData] = useState<MyReviewsData[] | null>([]);
+
+	const { getSession } = useAuth();
+	const [opened, { open: openSessionModal, close: closeSessionModal }] = useDisclosure(false);
+
+	useEffect(() => {
+		if (!sessionData) {
+			setIsLoading(true);
+			getSession()
+				.then((session) => {
+					setSessionData(session);
+					if (!session) openSessionModal();
+				})
+				.catch(console.error)
+				.finally(() => {
+					setIsLoading(false);
+				});
+		}
+	}, [getSession, openSessionModal, sessionData]);
+
+	useEffect(() => {
+		if (!sessionData) {
+			return;
+		}
+		setIsLoading(true);
+
+		const fetchMyReviews = async (): Promise<void> => {
+			try {
+				const results = await fetcher<{ data: MyReviewsData[] }>(
+					`${BASE_API_ENDPOINT}/myreviews`,
+					sessionData.access_token || ""
+				);
+
+				setMyReviewsData(results.data?.data || []);
+			} catch (err) {
+				console.error(err);
+				// Handle error here
+			} finally {
+				setIsLoading(false);
+			}
+		};
```

Comment:
can this somehow be extracted into https://github.com/stamford-syntax-club/course-compose/blob/main/apps/web/lib/api/api.ts so we can write some tests in the future? (not now ofc we're all lazy XD)

Reply:
Should I just stick with this instead of [fetcher](https://github.com/stamford-syntax-club/course-compose/blob/main/apps/web/utils/fetcher.ts) ?